### PR TITLE
Fixed incorrect static class field initialization order

### LINKED ISSUE #6493: incorrect static class field initialization order
> Choose one: is this a bug report or feature request?

bug report

### Input Code

```js
function one() {
  console.log("one");
  return "one";
}

function one_init() {
  console.log("one init");
  return 1;
}

function two() {
  console.log("two");
  return "two"
}

function two_init() {
  console.log("two init");
  return 2;
}

class X {
  static [one()] = one_init();
  static [two()] = two_init();
}

new X;
```

### Babel Configuration (.babelrc, package.json, cli command)

stage2, stage3

### Expected Behavior
one
two
one init
two init

### Current Behavior

one 
one init
two
two init

### Your Environment
[REPL](https://babeljs.io/repl/build/master#?babili=false&browsers=&build=&builtIns=false&code_lz=GYVwdgxgLglg9mABAgpgCgJSIN4ChGIQIDOcANigHRlwDmaARKgxgNz6IBOKUInSTMCgbsAvrlyhIsBMiEB9GGBhRMODkTCkK1Oo1SIlKluwLde_RAEYxEqdHhIoAdzhq8BTdqo16DF3AmHOZ8AgEMuOKS4A6yAYrKqlgehCTkPnr-roaJQWY8oYgATLa4EGQAhsTEiAAa6gTEUBWwEIgA2qiYALqIALxyKAkqmKaITS0wbe0BPf2I8UZJpULOdaxAA&debug=false&circleciRepo=&evaluate=true&lineWrap=false&presets=stage-2%2Cstage-3&targets=&version=7.0.0-beta.3)

### PR DESCRIPTION
**Pull request by Andarist**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #6493
| Patch: Bug Fix?          | yes
| Major: Breaking Change?  | a little?
| Minor: New Feature?      | no
| Tests Added + Pass?      | Yes
| Documentation PR         | no
| Any Dependency Changes?  | internal

### CODE REVIEW

File: packages/babel-plugin-transform-class-properties/src/index.js [hunk 1]
```diff
@@ -1,6 +1,27 @@
 import nameFunction from "@babel/helper-function-name";
 import template from "@babel/template";
 import syntaxClassProperties from "@babel/plugin-syntax-class-properties";
+import * as t from "@babel/types";
+
+const isSafeComputedKey = (node, scope) => {
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-class-properties/src/index.js]: This could go in `NodePath` (something like `path.isConstantExpression()`).

**Comment by gsathya** [comment]
This doesn't work correctly with regular computed properties -- [REPL](https://babeljs.io/repl/build/5447/#?babili=false&browsers=&build=&builtIns=false&code_lz=GYVwdgxgLglg9mABAgpgCgJSIN4ChGIQIDOcANigHRlwDmaARKgxgNz6IBOKUInSTMCgbsAvrlyhIsBMiEB9GGBhRMODkTCkK1Oo1SIlKluwLde_RAEYxEqdHhIoAdzhq8BTdqo16DF3AmHOZ8AgEMuOKS4A6yAYrKqlgehCTkPnr-roaJQWY8oYgATLbR0o5yKAAOcEpQwDAAbujJGmk6vvpCNXUNzXlcBZaC1bVg9U3CpRBkAIbExIgAGuoExFCzsBCIANqomAC6iAC8lQkqmKa7qD3jfS0H7oMWSDaI4msbW7sBhyeI8SMSVKQmcyyAA&debug=false&circleciRepo=&evaluate=true&lineWrap=false&presets=stage-3&targets=&version=7.0.0-beta.3%2Bpr.6530) should print 
```
one
onepointfive
two
one init
two init
```

**Comment by Andarist** [comment]
@gsathya thanks for reporting. This is a cross-plugin issue and truth to be told I haven't yet learned how to deal with those, as plugins should stay independent.

**Comment by Andarist** [comment]
@nicolo-ribaudo ive updated the PR, second pass of code review would be appreciated

@gsathya fixed :) can be checked on [updated repl](https://babeljs.io/repl/build/5545/#?babili=false&browsers=&build=&builtIns=false&code_lz=GYVwdgxgLglg9mABAgpgCgJSIN4ChGIQIDOcANigHRlwDmaARKgxgNz6IBOKUInSTMCgbsAvrlyhIsBMiEB9GGBhRMODkTCkK1Oo1SIlKluwLde_RAEYxEqdHhIoAdzhq8BTdqo16DF3AmHOZ8AgEMuOKS4A6yAYrKqlgehCTkPnr-roaJQWY8oYgATLbR0o5yKAAOcEpQwDAAbujJGmk6vvpCNXUNzXlcBZaC1bVg9U3CpRBkAIbExIgAGuoExFCzsBCIANqomAC6iAC8lQkqmKa7qD3jfS0H7oMWSDaI4msbW7sBhyeI8SMSVKQmcyyAA&debug=false&circleciRepo=&evaluate=true&lineWrap=false&presets=stage-3&targets=&version=7.0.0-beta.3%2Bpr.6530)

**Comment by gsathya** [comment]
> @gsathya fixed :) can be checked on updated repl

💯 awesome work!

File: packages/babel-plugin-proposal-class-properties/test/fixtures/general/computed-initialization-order/exec.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  set [track(4)](value) {}
+  [track(5)] = track(0);
+  static [track(6)] = track(11);
+  static [track(7)] = track(12);
+  [track(8)]() {}
+  [track(9)] = track(0);
+}
+
+const expected = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+assert.deepEqual(actual, expected);
```
Comment (xtuc) [inline packages/babel-plugin-proposal-class-properties/test/fixtures/general/computed-initialization-order/exec.js]: This test is very good.

File: packages/babel-traverse/src/path/introspection.js [hunk 3]
```diff
@@ -398,3 +398,44 @@ export function _resolve(dangerous?, resolved?): ?NodePath {
     }
   }
 }
+
+export function isConstantExpression() {
```
Comment (hzoo) [inline packages/babel-traverse/src/path/introspection.js]: I know this is different than evaluate but do we have to do this for other kinds of nodes? https://github.com/babel/babel/blob/master/packages/babel-traverse/src/path/evaluation.js#L78

Comment (Andarist) [reply packages/babel-traverse/src/path/introspection.js]: @hzoo i probably have skipped some kinds of nodes, just included what i could think at the moment - thanks for the pointer to the `_evaluate`, I might expand this `isConstantExpression` (btw is it good name?) later with info found in the `_evaluate`

**Pull request merged**